programs/monaco_protocol/src/state/market_liquidities.rs

```diff
@@ -0,0 +1,414 @@
+use std::cmp::Ordering;
+use std::string::ToString;
+
+use anchor_lang::prelude::*;
+
+use crate::error::CoreError;
+use crate::state::type_size::*;
+
+#[account]
+pub struct MarketLiquidities {
+    pub market: Pubkey,
+    pub liquidities_for: Vec<MarketOutcomePriceLiquidity>,
+    pub liquidities_against: Vec<MarketOutcomePriceLiquidity>,
+}
+
+impl MarketLiquidities {
+    pub const SIZE: usize = DISCRIMINATOR_SIZE
+        + PUB_KEY_SIZE // market
+        + vec_size(18_usize, 0_usize) // liquidities_for
+        + vec_size(18_usize, 0_usize); // liquidities_against
+
+    pub fn get_liquidity_for(&self, outcome: u16, price: f64) -> MarketOutcomePriceLiquidity {
+        match self
+            .liquidities_for
+            .binary_search_by(Self::sorter_for(outcome, price))
+        {
+            Ok(index) => self.liquidities_for[index],
+            Err(_) => MarketOutcomePriceLiquidity {
+                outcome,
+                price,
+                liquidity: 0,
+            },
+        }
+    }
+
+    pub fn get_liquidity_against(&self, outcome: u16, price: f64) -> MarketOutcomePriceLiquidity {
+        match self
+            .liquidities_against
+            .binary_search_by(Self::sorter_for(outcome, price))
+        {
+            Ok(index) => self.liquidities_for[index],
+            Err(_) => MarketOutcomePriceLiquidity {
+                outcome,
+                price,
+                liquidity: 0,
+            },
+        }
+    }
```

Comment:
```suggestion
    pub fn get_liquidity_against(&self, outcome: u16, price: f64) -> MarketOutcomePriceLiquidity {
        match self
            .liquidities_against
            .binary_search_by(Self::sorter_against(outcome, price))
        {
            Ok(index) => self.liquidities_against[index],
            Err(_) => MarketOutcomePriceLiquidity {
                outcome,
                price,
                liquidity: 0,
            },
        }
    }
```
That's what you get for skipping unit tests!

Reply:
yes